spring-cloud-task-batch/src/main/java/org/springframework/cloud/task/batch/configuration/TaskJobLauncherApplicationRunnerFactoryBean.java

```
@@ -0,0 +1,106 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.task.batch.configuration;
+
+import java.util.List;
+
+import org.springframework.batch.core.Job;
+import org.springframework.batch.core.configuration.JobRegistry;
+import org.springframework.batch.core.explore.JobExplorer;
+import org.springframework.batch.core.launch.JobLauncher;
+import org.springframework.batch.core.repository.JobRepository;
+import org.springframework.beans.factory.FactoryBean;
+import org.springframework.boot.autoconfigure.batch.BatchProperties;
+import org.springframework.cloud.task.batch.handler.TaskJobLauncherApplicationRunner;
+import org.springframework.util.Assert;
+import org.springframework.util.StringUtils;
+
+/**
+ * Factory bean for creating an instance of {@link TaskJobLauncherApplicationRunner}.
+ *
+ * @author Glenn Renfro
+ * @since 2.3.0
+ */
+public class TaskJobLauncherApplicationRunnerFactoryBean
+		implements FactoryBean<TaskJobLauncherApplicationRunner> {
+
+	private JobLauncher jobLauncher;
+
+	private JobExplorer jobExplorer;
+
+	private List<Job> jobs;
+
+	private String jobNames;
+
+	private JobRegistry jobRegistry;
+
+	private Integer order = 0;
+
+	private TaskBatchProperties taskBatchProperties;
+
+	private JobRepository jobRepository;
+
+	public TaskJobLauncherApplicationRunnerFactoryBean(JobLauncher jobLauncher,
+			JobExplorer jobExplorer, List<Job> jobs,
+			TaskBatchProperties taskBatchProperties, JobRegistry jobRegistry,
+			JobRepository jobRepository, BatchProperties batchProperties) {
+		Assert.notNull(taskBatchProperties, "taskBatchProperties must not be null");
+		Assert.notNull(batchProperties, "batchProperties must not be null");
+		this.jobLauncher = jobLauncher;
+		this.jobExplorer = jobExplorer;
+		Assert.notEmpty(jobs, "jobs must not be null nor empty");
```

Move asserts together to fail fast.